manager/integration/tests/test_csi.py

```diff
@@ -489,20 +495,72 @@ def test_csi_mount_volume_online_expansion(client, core_api, storage_class, pvc,
 
     1. Create a new `storage_class` with `allowVolumeExpansion` set
     2. Create PVC and Pod with the new StorageClass
-    3. Generate `test_data` in a directory the volume is not mounted on.
-    4. Copy the `test data` to the volume mount point asynchronously.
-    5. During the copy, update pvc.spec.resources to expand the volume
-    6. Verify the volume expansion done using Longhorn API
-      and Check the PVC & PV size
-    7. Wait for the copy complete.
-    8. Calculate the checksum for the copied data inside the volume mount point
-      asynchronously.
-    9. During the calculation, update pvc.spec.resources to expand the volume
-      again.
-    10. Wait for the calculation complete, then compare the checksum.
-    11. Do cleanup: Remove the original `test_data`as well as the pod and PVC.
+    3. Use `dd` command copy data into volume mount point.
+    4. During the copy, update pvc.spec.resources to expand the volume
+    5. Verify the volume expansion done using Longhorn API
+       and Check the PVC & PV size
+    6. Wait for the copy complete.
+    7. Calculate the checksum for the copied data inside volume mount point.
+    8. Update pvc.spec.resources to expand the volume and get data checksum
+       again.
+    9. Wait for the calculation complete, then compare the checksum.
+    10. Do cleanup: Remove the original `test_data`as well as the pod and PVC.
     """
-    pass
+    create_storage_class(storage_class)
+    volume_data_path = "/data/file"
+
+    pod_name = "csi-mount-volume-online-expansion-test"
+    pvc_name = pod_name + "-pvc"
+    pvc['metadata']['name'] = pvc_name
+    pvc['spec']['storageClassName'] = storage_class['metadata']['name']
+    create_pvc(pvc)
+
+    pod_manifest['metadata']['name'] = pod_name
+    pod_manifest['spec']['volumes'] = [create_pvc_spec(pvc_name)]
+    pod_manifest['spec']['nodeName'] = get_self_host_id()
+    create_and_wait_pod(core_api, pod_manifest)
+
+    volume_name = get_volume_name(core_api, pvc_name)
+
+    # Use new process to do copy data and expand at the same time
+    cmd = ["kubectl", "exec", "-it", pod_name, "--", "/bin/sh", "-c",
+           "dd if=/dev/urandom of={} bs=1M count=2500".
+           format(volume_data_path)]
+    process = subprocess.Popen(cmd)
+
+    expand_size = str(EXPANDED_VOLUME_SIZE*Gi)
+    expand_and_wait_for_pvc(core_api, pvc, EXPANDED_VOLUME_SIZE*Gi)
+    wait_for_volume_expansion(client, volume_name)
+    volume = client.by_id_volume(volume_name)
+    assert volume.size == expand_size
+
```

Comment:
Why not below? some code seems redundant because some assert check already in wait function.

```python
    expand_size = str(EXPANDED_VOLUME_SIZE*Gi)
    expand_and_wait_for_pvc(core_api, pvc, expand_size)
    wait_for_volume_expansion(client, volume_name, expand_size)

```